packages/frontend/src/components/nft/NFTDetailWrapper.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+
+import { selectAccountId, selectBalance } from "../../redux/slices/account";
+import {
+    actions as nftActions,
+    selectTokenForAccountForContractForTokenId,
+    selectTransferredTokenForContractForTokenId,
+} from '../../redux/slices/nft';
+import { NFTDetail } from "./NFTDetail";
+
+export function NFTDetailWrapper ({
+    match,
+    history
+}) {
+    const { contractId: contractName, tokenId } = match.params;
+    const accountId = useSelector(selectAccountId);
+    const { balanceAvailable: nearBalance} = useSelector(selectBalance);
+    const nft = useSelector((state) => selectTokenForAccountForContractForTokenId(state, {
+        accountId,
+        contractName,
+        tokenId,
+    }));
+
+    const transferredNft = useSelector((state) => selectTransferredTokenForContractForTokenId(state, {
```

Comment:
This is the biggest change from the original PR - rather than fetching the NFT metadata by `tokenId` (without regards to `accountId`) I've added two new selectors to fetch the NFT state:
- `selectTokenForAccountForContractForTokenId` gets the NFT metadata from state initialized when `fetchNFTs` was dispatched.
- `selectTransferredTokenForContractForTokenId` gets NFT metadata from a new key in the redux state that holds the set of tokens the user has transferred. Once transferred, the NFT no longer exists in state as an owned token since it is no longer owned by the user. Rather than refetch the NFT after transferring I've opted to simply add the transferred token to the new state. Refetching the token is still an option however and only requires a small change to the new `transferNFT` action/reducer.